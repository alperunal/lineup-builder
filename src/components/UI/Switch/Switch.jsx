import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

import './Switch.scss';

const Switch = ({ id, checked, size, handleChange }) => (
  <label className="switch" htmlFor={`switch-${id}`}>
    <input
      type="checkbox"
      id={`switch-${id}`}
      checked={checked}
      onChange={handleChange}
    />
    <span
      className={cn('slider', 'round', {
        'slider--small': size === 'small',
        'slider--medium': size === 'medium',
        'slider--large': size === 'large',
      })}
    />
  </label>
);

Switch.propTypes = {
  id: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  size: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default Switch;
