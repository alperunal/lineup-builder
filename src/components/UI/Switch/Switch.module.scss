.switch {
  position: relative;
  display: inline-block;
  width: 30px;
  height: 17px;

  input {
    opacity: 0;
    width: 0;
    height: 0;
  }
}

.slider {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;

  &:before {
    position: absolute;
    content: '';
    height: 14px;
    width: 14px;
    left: 2px;
    bottom: 2px;
    background-color: white;
    -webkit-transition: 0.4s;
    transition: 0.4s;
  }

  &.round {
    border-radius: 34px;

    &:before {
      border-radius: 50%;
    }
  }
}

input:checked + .slider {
  background-color: var(--color-primary);
}

input:focus + .slider {
  box-shadow: 0 0 1px var(--color-primary);
}

input:checked + .slider:before {
  -webkit-transform: translateX(12px);
  -ms-transform: translateX(12px);
  transform: translateX(12px);
}
