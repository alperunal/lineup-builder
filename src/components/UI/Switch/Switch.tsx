import React from 'react';
import cn from 'classnames';
import './Switch.module.scss';

type Props = {
    id: string,
    checked: boolean,
    size?: 'large' | 'medium' | 'small',
    handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
};

const Switch = ({
    id, checked, size, handleChange,
}: Props): JSX.Element => (
    <label className="switch" htmlFor={`switch-${id}`}>
        <input
            type="checkbox"
            id={`switch-${id}`}
            checked={checked}
            onChange={handleChange}
        />
        <span
            className={
                cn(
                    'slider',
                    'round',
                    {
                        'slider--small': size === 'small',
                        'slider--medium': size === 'medium',
                        'slider--large': size === 'large',
                    },
                )
            }
        />
    </label>
);

export default Switch;
