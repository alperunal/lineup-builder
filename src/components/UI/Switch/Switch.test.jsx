import React from 'react';
import { shallow } from 'enzyme';
import Switch from './Switch';

it('should render Switch correctly', () => {
  const wrapper = shallow(
    <Switch id="test" checked={false} handleChange={jest.fn()} />,
  );
  expect(wrapper).toMatchSnapshot();
});
