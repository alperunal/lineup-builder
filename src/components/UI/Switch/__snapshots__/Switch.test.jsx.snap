// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Switch correctly 1`] = `ShallowWrapper {}`;
